components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/core/management/PreUpdatePasswordActionDTOModelResolver.java

```diff
@@ -0,0 +1,274 @@
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.ActionDTO;
+import org.wso2.carbon.identity.action.management.service.ActionDTOModelResolver;
+import org.wso2.carbon.identity.certificate.management.exception.CertificateMgtClientException;
+import org.wso2.carbon.identity.certificate.management.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.model.Certificate;
+import org.wso2.carbon.identity.certificate.management.service.CertificateManagementService;
+import org.wso2.carbon.identity.user.pre.update.password.action.internal.PreUpdatePasswordActionServiceComponentHolder;
+import org.wso2.carbon.identity.user.pre.update.password.action.service.model.PasswordSharing;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.user.pre.update.password.action.core.constant.PreUpdatePasswordActionConstants.CERTIFICATE;
+import static org.wso2.carbon.identity.user.pre.update.password.action.core.constant.PreUpdatePasswordActionConstants.PASSWORD_SHARING_FORMAT;
+
+/**
+ * This class implements the methods required to resolve ActionDTO objects in Pre Update Password extension.
+ */
+public class PreUpdatePasswordActionDTOModelResolver implements ActionDTOModelResolver {
+
+    @Override
+    public Action.ActionTypes getSupportedActionType() {
+
+        return Action.ActionTypes.PRE_UPDATE_PASSWORD;
+    }
+
+    @Override
+    public ActionDTO resolveForAddOperation(ActionDTO actionDTO, String tenantDomain)
+            throws ActionDTOModelResolverException {
+
+        Map<String, Object> properties = new HashMap<>();
+        // Certificate is an optional field.
+        Object certificate = actionDTO.getProperty(CERTIFICATE);
+        if (certificate != null) {
+            if (!(certificate instanceof Certificate)) {
+                throw new ActionDTOModelResolverClientException("Invalid Request",
+                        "Certificate should be an instance of Certificate.");
+            }
+
+            Certificate certToBeAdded = buildCertificate(actionDTO.getId(), (Certificate) certificate);
+            String certificateId = addCertificate(certToBeAdded, tenantDomain);
+            properties.put(CERTIFICATE, certificateId);
+        }
+
+        // Password sharing format is a required field.
+        if (actionDTO.getProperty(PASSWORD_SHARING_FORMAT) == null) {
+            throw new ActionDTOModelResolverClientException("Invalid Request",
+                    "Password sharing format is a required field.");
+        }
+        if (!(actionDTO.getProperty(PASSWORD_SHARING_FORMAT) instanceof PasswordSharing.Format)) {
+            throw new ActionDTOModelResolverClientException("Invalid Request",
+                    "Password sharing format should be an instance of PasswordSharing.Format.");
```

Comment:
instance of PasswordSharing.Format is an implementation detail. Do not expose that.
Say something like unsupported password sharing format

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6281/commits/604c0addbbf444856b2f38e31bf645ccf07965b4